c/src/lib.rs

```diff
@@ -794,3 +794,13 @@ pub extern "C" fn substrait_validator_export_proto(
 ) -> *mut u8 {
     export(substrait_validator::export::Format::Proto, handle, size)
 }
+
+/// Returns the version of Substrait that the validator was built against.
+#[no_mangle]
+pub extern "C" fn substrait_validator_substrait_version() -> *const libc::c_char {
+    static VERSION: once_cell::sync::OnceCell<std::ffi::CString> = once_cell::sync::OnceCell::new();
+    VERSION
+        .get_or_init(|| std::ffi::CString::new(substrait_validator::substrait_version()).unwrap())
+        .as_bytes_with_nul()
+        .as_ptr() as *const libc::c_char
+}
```

Comment:
If you go with `git-version` then you should be able to assign the `const` version to a cell.

Also, why doesn't [`std::sync::Once`](https://doc.rust-lang.org/stable/std/sync/struct.Once.html) work here?

Reply:
> Also, why doesn't [`std::sync::Once`](https://doc.rust-lang.org/stable/std/sync/struct.Once.html) work here?

Looks like [that would require an `unsafe` block and a mutable static](https://doc.rust-lang.org/stable/std/sync/struct.Once.html#examples-1)? I imagine that `once_cell` just abstracts away from that.

`once_cell` is already a transitive dependency (the core lib requires it), so using it doesn't really cost anything.